src/stest/src/main.rs

```diff
@@ -0,0 +1,19 @@
+use clap::Parser;
+use std::io::{stdin, stdout};
+use std::process::ExitCode;
+use stest::App;
+use stest::config::Config;
+
+/// The stest program filters a list of files by their properties, in a way that is analogous to
+/// bash's builtin test, except that files that pass all tests are printed to stdout.
+fn main() -> ExitCode {
+    let config = Config::parse();
+    let mut stdin = stdin().lock();
+    let mut stdout = stdout().lock();
+    let result = App::new(config)
+        .run(&mut stdin, &mut stdout);
+    match result {
+        Ok(_) => ExitCode::SUCCESS,
+        Err(_) => ExitCode::FAILURE,
+    }
```

Comment:
If you made all of the tests return `Result`, here is where you would print out the error before exiting. You'd just `Ok(true)` to success, `Ok(false)` to failure, and `Err(e)` to print out the error and return failure.